src/nap.js

```diff
@@ -225,7 +227,7 @@ function newWeb(){
 
   web.use = function() {
     if(!arguments.length) return web
-    middleware = toArray(arguments).reverse().concat(middleware)
+    middleware.unshift.apply(middleware, toArray(arguments).reverse())
```

Comment:
The difference is, that the middleware list is kept up-to-date during the whole lifecycle, and added middlewares are called for each handler, whenever the handles is called, and not constructed when the resource is added.